modules/apps/transfer/ibc_module_test.go

```diff
@@ -239,3 +240,75 @@ func (suite *TransferTestSuite) TestOnChanOpenAck() {
 		})
 	}
 }
+
+func (suite *TransferTestSuite) TestOnChanUpgradeInit() {
+	var path *ibctesting.Path
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expError error
+	}{
+		{
+			"success",
+			func() {}, // successful happy path for a standalone transfer app is swapping out the underlying connection
+			nil,
+		},
+		{
+			"invalid upgrade connection",
```

Comment:
In theory we don't need this test here, right? Because the test will fail [here](https://github.com/cosmos/ibc-go/blob/347346b51c9a2bec220f85c2406e021311ede5b4/modules/core/04-channel/keeper/upgrade.go#L778) before reaching the transfer callback, right?

Reply:
Correct, this is failing well before the application is ever called, I'd be fine with removing it as there is already tests using the mock app in core for this. I just put it here to make things whole (testing version, ordering and connection)

Reply:
I'll leave this for now, we can remove later if needs be